File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/teleport/RandomTeleportSettingsService.java
package top.likoslupus.cellulosesz.api.teleport;

import java.util.concurrent.CompletableFuture;

public interface RandomTeleportSettingsService {

    RandomTeleportSettings settings(String world);

    CompletableFuture<Void> setCenter(String world, double x, double z);

    CompletableFuture<Void> setMinimumRadius(String world, int radius);

    CompletableFuture<Void> setMaximumRadius(String world, int radius);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/teleport/RandomTeleportSettingsService.kt
package top.likoslupus.cellulosesz.api.teleport

import java.util.concurrent.CompletableFuture

public interface RandomTeleportSettingsService {

    public fun settings(world: String): RandomTeleportSettings

    public fun setCenter(world: String, x: Double, z: Double): CompletableFuture<Void>

    public fun setMinimumRadius(world: String, radius: Int): CompletableFuture<Void>

    public fun setMaximumRadius(world: String, radius: Int): CompletableFuture<Void>

}
